torch/autograd/gradcheck.py

```diff
@@ -1164,7 +1164,7 @@ def _vec_from_tensor(x, generator, downcast_complex=False):
         dtype = _to_real_dtype(x.dtype) if downcast_complex else x.dtype
         values = torch.rand(x_values.numel(), generator=generator) \
             .to(dtype=dtype, device=x.device) \
-            .reshape(x_values.shape)
+            .view(x_values.shape)
```

Comment:
Why don't we pass in `x_values.shape` to `torch.rand()` directly? :p

Reply:
I have no idea! Maybe they wanted to generate a view?

Reply:
While we're on the topic, why don't we just generate this on the correct device in the first place lol

Reply:
@soulitzer might remember. My guess is that this is a combination of two things: we use a custom generator to not mess up user rng and so we always generate on CPU and this is a copy-paste from the non-sparse code below haha

Reply:
You can def ignore and merge as-is.